### ISSUE TITLE
send message from server to specific user

### ISSUE THREAD

**Issue by ovoshlook** [openresty/lua-resty-websocket#14]
Hi is websocket have any way to send message from server to specific user
Like ws:send_text("some text",receiver_ip)?

**Comment by bungle** [comment]
Each web socket connection is for specific user. And your server has these handles. You could set up a queue based on say REMOTE_ADDR etc. But I would use something that is unique to user, you see many users can share the same IP behind the NAT.

**Comment by ovoshlook** [comment]
Thanks for reply.
So am i right that i can get somewhere source IP of each user that  connecting to websocket server
like ngx.source_ip() or something like this?

**Comment by bungle** [comment]
@ovoshlook, of course, it is normal HTTP header called `ngx.header.remote_addr`, but I wouldn't use it. You should at least check for reverse proxies, e.g.: https://github.com/bungle/lua-resty-session/blob/master/lib/resty/session.lua#L235-L252 (session module btw. doesn't check address by default because of many problems it might cause - but in a controlled environment, it is okay).

**Comment by ouyangruoxue** [comment]
I have the same problem ,i use ngx.semaphore but it don't work, Can you tell me your solution ? @ovoshlook

**Comment by ouyangruoxue** [comment]
使用ngx.semaphore可以完成对应功能
